### ISSUE TITLE
provide more context to help GPT understand the meaning of pandas data structure, for example, more description for each column

### ISSUE THREAD

**Issue by ihorizons2022** [Sinaptik-AI/pandas-ai#312]
### 🐛 Describe the bug

is there any method to achieve this except modify the prompt

**Comment by gventuri** [comment]
That sounds like a very nice addition. Will work on it asap!
Thanks for reporting @ihorizons2022.

If in the meanwhile you have examples of context that you needed to pass in the past, feel free to share, it could be incredibly beneficial as we try to implement the feature.

**Comment by SpookkyMaDookky** [comment]
Ditto.
An example would be... I have a column titled "cadence" but the user might ask about "frequency" instead. In the past I have written longer descriptive prompts which provide details of each column. The easiest way to do this might be for the ability to add more descriptive prompts to the base prompt in `generate_python_code` as an additional argument?

**Comment by RobertHH-IS** [comment]
A must have. Ability to describe each column - alternative wording - and input examples using vector store of examples when it comes to more abstract questions.

**Comment by cjviper** [comment]
What about sending the `dtype` of the columns (automatically)? Currently only the head of the DataFrame is being sent to the LLM so the LLM has to infer the dtypes. This can be a problem when numbers are stored in string columns. It should at least be optional to have pandas AI pass in the dtypes to prevent problems like the following:

```
df = pd.DataFrame({'item_id':['10','20','30'], 'a':[1, 2, 3]})
print(pandas_ai(df, prompt='What is the value of a for item_id 20?')) # fails
print(pandas_ai(df, prompt='What is the value of a for item_id 20? item_id is a string')) # succeeds
```

I'm not sure whether this request should be part of this issue or raised separately.

**Comment by aaryangrg** [comment]
There are some use cases which would also require the need to describe data across rows. For example, A list of dictionaries originally present in a column, now split into individual rows, maintaining all other data consistently, is something I had to do before using the library. A simple line explaining that data belonging to one data-point may be split across multiple rows (All having some common unique ID) helped during inference. 
Would be happy to help implement this.

**Comment by ihorizons2022** [comment]
glad to hear the feature will be included in v1.0
could you please share how to implement the feature?
besides, when version 1.0 will be released
